test/tools/ossfuzz/yulProto_diff_ossfuzz.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <fstream>
+
+#include <test/tools/ossfuzz/yulProto.pb.h>
+#include <test/tools/fuzzer_common.h>
+#include <test/tools/ossfuzz/protoToYul.h>
+#include <src/libfuzzer/libfuzzer_macro.h>
+
+#include <libyul/AssemblyStack.h>
+#include <liblangutil/EVMVersion.h>
+
+#include <test/tools/yulInterpreter/Interpreter.h>
+
+using namespace yul;
+using namespace yul::test::yul_fuzzer;
+using namespace std;
+
+using namespace langutil;
+using namespace dev;
+using namespace yul::test;
+
+void interpret(ostream& os, shared_ptr<yul::Block> ast)
+{
+	InterpreterState state;
+	state.maxTraceSize = 10000;
+	Interpreter interpreter(state);
+	try
+	{
+		interpreter(*ast);
+	}
+	catch (InterpreterTerminated const&)
+	{
+	}
+
+	os << "Trace:" << endl;
+	for (auto const& line: interpreter.trace())
+		os << "  " << line << endl;
+}
+
+DEFINE_BINARY_PROTO_FUZZER(Function const& _input)
+{
+	string yul_source = functionToString(_input);
+
+	if (const char* dump_path = getenv("PROTO_FUZZER_DUMP_PATH"))
+	{
+		// With libFuzzer binary run this to generate a YUL source file x.yul:
+		// PROTO_FUZZER_DUMP_PATH=x.yul ./a.out proto-input
+		ofstream of(dump_path);
+		of.write(yul_source.data(), yul_source.size());
+	}
+
+	// AssemblyStack entry point
+	AssemblyStack stack(langutil::EVMVersion::constantinople(), AssemblyStack::Language::StrictAssembly);
```

Comment:
Petersburg?

Reply:
I would say just use the default constructor.

Reply:
changed to default constructor